root/static/scripts/edit/components/ArtistCreditBubble.js

```diff
@@ -0,0 +1,105 @@
+// This file is part of MusicBrainz, the open internet music database.
+// Copyright (C) 2016 MetaBrainz Foundation
+// Licensed under the GPL version 2, or (at your option) any later version:
+// http://www.gnu.org/licenses/gpl-2.0.txt
+
+const $ = require('jquery');
+const {assign} = require('lodash');
+const React = require('react');
+
+const ArtistCreditLink = require('../../common/components/ArtistCreditLink');
+const DescriptiveLink = require('../../common/components/DescriptiveLink');
+const {l} = require('../../common/i18n');
+const {reduceArtistCredit} = require('../../common/immutable-entities');
+const clean = require('../../common/utility/clean');
+const ArtistCreditNameEditor = require('./ArtistCreditNameEditor');
+
+function onBubbleKeyDown(done, hide, event) {
+  if (event.isDefaultPrevented()) {
+    return;
+  }
+  const pressedEsc = event.which === 27;
+  const pressedEnter = event.which === 13;
+
+  if (pressedEsc || (pressedEnter && $(event.target).is(':not(:button)'))) {
+    event.preventDefault();
+    if (pressedEsc) {
+      hide();
+    } else if (pressedEnter) {
+      done();
+    }
+  }
+}
+
+const ArtistCreditBubble = ({
+  addName,
+  artistCredit,
+  copyArtistCredit,
+  done,
+  entity,
+  extraButtons,
+  extraContent,
+  hide,
+  onNameChange,
+  pasteArtistCredit,
+  removeName,
+}) => (
```

Comment:
The `({addName, artistCredit, ...})` part is the "destructuring" feature, yeah. It's the same feature as `const {l} = ...` above, except it's implicitly assigned from the function argument. And the `=>` is just arrow functions.